controllers/utils.js

```diff
@@ -33,39 +35,57 @@ function hasUserSignedUpForRelay(user) {
 
 function getExperimentBranch(req, sorterNum) {
 
+  if (req.session.excludeFromExperiment && !req.query.experimentBranch) {
+    log.debug("This session has already been excluded from the experiment");
+    return false;
+  }
+
   // If we cannot parse req.headers["accept-language"], we should not
   // enroll users in the experiment.
   if (!req.headers || !req.headers["accept-language"]){
+    log.debug("No headers or accept-language information present.");
     return false;
   }
 
   // If the user doesn't have an English variant langauge selected as their primary language,
   // we do not enroll them in the experiment.
   const lang = req.headers["accept-language"].split(",");
   if (!lang[0].includes("en")) {
+    log.debug("Preferred language is not English variant: ", lang[0]);
     return false;
   }
 
   // If URL param has experimentBranch entry, use that branch;
   if (req.query.experimentBranch) {
     if (!["va", "vb"].includes(req.query.experimentBranch)) {
+      log.debug("The requested branch is unknown: ", req.query.experimentBranch);
       return false;
     }
+    log.debug("This session has been set to the requested branch: ", req.query.experimentBranch);
+    req.session.excludeFromExperiment = false;
```

Comment:
Because this session variable is only set during `return false` occasions, this reset here was necessary.